Update tests used to check version

<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Also test if the latest release and devel branch are affected too -->
<!--- Complete *all* sections as described, this form is processed automatically -->

##### SUMMARY
In the testing pipeline, we have several tests across the roles that look at what version was installed.  Right now we're hard-coding the version based on the distribution rather than looking at the variable that is being set as part of the testing pipeline.  Need to update the test logic to look at the version that we're telling it to (via the `zabbix_<component>_version` variable and base the version on that insead.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
All roles

##### ACTUAL RESULTS
```

=================================== FAILURES ===================================
_____________ test_zabbix_package[ansible://zabbix-agent-centos8] ______________

host = <testinfra.host.Host ansible://zabbix-agent-centos8>
zabbix_agent_package = <package zabbix-agent2>

    def test_zabbix_package(host, zabbix_agent_package):
        assert zabbix_agent_package.is_installed

        if host.system_info.distribution == "debian":
            if host.system_info.codename in ["bullseye", "focal"]:
                assert zabbix_agent_package.version.startswith("1:6.4")
            else:
>               assert zabbix_agent_package.version.startswith("1:6.0")
E               AssertionError: assert False
E                +  where False = <built-in method startswith of str object at 0x7f46404dbc10>('1:6.0')
E                +    where <built-in method startswith of str object at 0x7f46404dbc10> = '1:6.4.1-1+debian10'.startswith
E                +      where '1:6.4.1-1+debian10' = <package zabbix-agent2>.version

molecule/agent2/tests/common/test_agent.py:47: AssertionError
=========================== short test summary info ============
```

--- a/molecule/zabbix_proxy/tests/test_default.py
+++ b/molecule/zabbix_proxy/tests/test_default.py
@@ -17,40 +17,20 @@ def test_zabbixproxy_running_and_enabled(host):
         assert zabbix.is_running
 
 
-@pytest.mark.parametrize(
-    "proxy", [("zabbix-proxy-pgsql"), ("zabbix-proxy-mysql"), ("zabbix-proxy-sqlite3")]
-)
-def test_zabbix_package(host, proxy):
+def test_zabbix_package(host):
     ansible_data = host.ansible.get_variables()
-    zabbixhost = ansible_data["inventory_hostname"]
+    version = ansible_data['zabbix_proxy_version']
+    database = ansible_data['zabbix_proxy_database']
 
-    zabbixhost = zabbixhost.replace("-centos", "")
-    zabbixhost = zabbixhost.replace("-debian", "")
-    zabbixhost = zabbixhost.replace("-ubuntu", "")
-
-    if zabbixhost == proxy:
-        zabbix_proxy = host.package(proxy)
-        if host.system_info.distribution in ["debian", "ubuntu"]:
-            assert zabbix_proxy.version.startswith("1:6.4")
-        elif host.system_info.distribution == "centos":
-            assert zabbix_proxy.version.startswith("6.4")
-        assert zabbix_proxy.is_installed
+    zabbix_proxy = host.package(f'zabbix-proxy-%s' % database)
+    assert str(version) in zabbix_proxy.version
 
 
 def test_zabbix_proxy_dot_conf(host):
-    found = False
-    for file_name in [
-        "/etc/zabbix/zabbix_proxy.conf",
-        "/etc/zabbix_proxy.conf",
-    ]:
-        if host.file(file_name).exists:
-            found = True
-            break
-
-    assert found
-    zabbix_proxy_conf = host.file(file_name)
-    assert zabbix_proxy_conf.user in ["zabbix", "zabbixsrv"]
-    assert zabbix_proxy_conf.group in ["zabbix", "zabbixsrv"]
+    zabbix_proxy_conf = host.file("/etc/zabbix/zabbix_proxy.conf")
+    assert zabbix_proxy_conf.exists
+    assert zabbix_proxy_conf.user == "zabbix"
+    assert zabbix_proxy_conf.group == "zabbix"
     assert zabbix_proxy_conf.mode == 0o644
 
     assert zabbix_proxy_conf.contains("ListenPort=10051")
@@ -60,24 +40,13 @@ def test_zabbix_proxy_dot_conf(host):
 def test_zabbix_include_dir(host):
     zabbix_include_dir = host.file("/etc/zabbix/zabbix_proxy.conf.d")
     assert zabbix_include_dir.is_directory
-    assert zabbix_include_dir.user in ["zabbix", "zabbixsrv"]
-    assert zabbix_include_dir.group in ["zabbix", "zabbixsrv"]
-    # assert zabbix_include_dir.mode == 0o644
+    assert zabbix_include_dir.user == "zabbix"
+    assert zabbix_include_dir.group == "zabbix"
 
 
 def test_zabbix_proxy_logfile(host):
-    found = False
-    for file_name in [
-        "/var/log/zabbix/zabbix_proxy.log",
-        "/var/log/zabbixsrv/zabbix_proxy.log",
-    ]:
-        if host.file(file_name).exists:
-            found = True
-            break
-
-    assert found
-
-    zabbix_logfile = host.file(file_name)
+    zabbix_logfile = host.file("/var/log/zabbix/zabbix_proxy.log")
+    assert zabbix_logfile.exists
     assert not zabbix_logfile.contains("Access denied for user")
     assert not zabbix_logfile.contains("database is down: reconnecting")
     assert zabbix_logfile.contains("current database version")

--- a/molecule/zabbix_server/tests/test_default.py
+++ b/molecule/zabbix_server/tests/test_default.py
@@ -19,38 +19,20 @@ def test_zabbiserver_running_and_enabled(host):
         assert zabbix.is_running
 
 
-@pytest.mark.parametrize("server", [("zabbix-server-pgsql"), ("zabbix-server-mysql")])
-def test_zabbix_package(host, server):
+def test_zabbix_package(host):
     ansible_data = host.ansible.get_variables()
-    zabbixhost = ansible_data["inventory_hostname"]
+    version = ansible_data['zabbix_server_version']
+    database = ansible_data['zabbix_server_database']
 
-    zabbixhost = zabbixhost.replace("-centos", "")
-    zabbixhost = zabbixhost.replace("-debian", "")
-    zabbixhost = zabbixhost.replace("-ubuntu", "")
-
-    if zabbixhost == server:
-        zabbix_server = host.package(server)
-        if host.system_info.distribution in ["debian", "ubuntu"]:
-            assert zabbix_server.version.startswith("1:6.4")
-        elif host.system_info.distribution == "centos":
-            assert zabbix_server.version.startswith("6.4")
-        assert zabbix_server.is_installed
+    zabbix_server = host.package(f'zabbix-server-%s' % database)
+    assert str(version) in zabbix_server.version
 
 
 def test_zabbix_server_dot_conf(host):
-    found = False
-    for file_name in [
-        "/etc/zabbix/zabbix_server.conf",
-        "/etc/zabbix_server.conf",
-    ]:
-        if host.file(file_name).exists:
-            found = True
-            break
-
-    assert found
-    zabbix_server_conf = host.file(file_name)
-    assert zabbix_server_conf.user in ["zabbix", "zabbixsrv"]
-    assert zabbix_server_conf.group in ["zabbix", "zabbixsrv"]
+    zabbix_server_conf = host.file("/etc/zabbix/zabbix_server.conf")
+    assert zabbix_server_conf.exists
+    assert zabbix_server_conf.user == "zabbix"
+    assert zabbix_server_conf.group == "zabbix"
     assert zabbix_server_conf.mode == 0o640
 
     assert zabbix_server_conf.contains("ListenPort=10051")
@@ -60,24 +42,13 @@ def test_zabbix_server_dot_conf(host):
 def test_zabbix_include_dir(host):
     zabbix_include_dir = host.file("/etc/zabbix/zabbix_server.conf.d")
     assert zabbix_include_dir.is_directory
-    assert zabbix_include_dir.user in ["zabbix", "zabbixsrv"]
-    assert zabbix_include_dir.group in ["zabbix", "zabbixsrv"]
-    # assert zabbix_include_dir.mode == 0o644
+    assert zabbix_include_dir.user == "zabbix"
+    assert zabbix_include_dir.group == "zabbix"
 
 
 def test_zabbix_server_logfile(host):
-    found = False
-    for file_name in [
-        "/var/log/zabbix/zabbix_server.log",
-        "/var/log/zabbixsrv/zabbix_server.log",
-    ]:
-        if host.file(file_name).exists:
-            found = True
-            break
-
-    assert found
-
-    zabbix_logfile = host.file(file_name)
+    zabbix_logfile = host.file("/var/log/zabbix/zabbix_server.log")
+    assert zabbix_logfile.exists
     assert not zabbix_logfile.contains("Access denied for user")
     assert not zabbix_logfile.contains("database is down: reconnecting")
     assert zabbix_logfile.contains("current database version")

--- a/molecule/zabbix_web/tests/test_default.py
+++ b/molecule/zabbix_web/tests/test_default.py
@@ -8,28 +8,13 @@ testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
 ).get_hosts("all")
 
 
-@pytest.mark.parametrize(
-    "server, redhat, debian",
-    [
-        ("zabbix-server-pgsql", "zabbix-web-pgsql", "zabbix-frontend-php"),
-        ("zabbix-server-mysql", "zabbix-web-mysql", "zabbix-frontend-php"),
-    ],
-)
-def test_zabbix_package(host, server, redhat, debian):
-    host = host.backend.get_hostname()
-    host = host.replace("-centos7", "")
-    host = host.replace("-centos8", "")
-    host = host.replace("-debian", "")
-    host = host.replace("-ubuntu", "")
+def test_zabbix_package(host):
+    ansible_data = host.ansible.get_variables()
+    version = ansible_data['zabbix_web_version']
+    database = ansible_data['zabbix_server_database']
 
-    if host == server:
-        if host.system_info.distribution in ["debian", "ubuntu"]:
-            zabbix_web = host.package(debian)
-            assert zabbix_web.version.startswith("1:6.4")
-        elif host.system_info.distribution == "centos":
-            zabbix_web = host.package(redhat)
-            assert zabbix_web.version.startswith("6.4")
-        assert zabbix_web.is_installed
+    zabbix_web = host.package(f'zabbix-web-%s' % database)
+    assert str(version) in zabbix_web.version
 
 
 def test_zabbix_web(host):

--- a/molecule/zabbix_web/tests/test_default.py
+++ b/molecule/zabbix_web/tests/test_default.py
@@ -11,9 +11,9 @@ testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
 def test_zabbix_package(host):
     ansible_data = host.ansible.get_variables()
     version = ansible_data['zabbix_web_version']
-    database = ansible_data['zabbix_server_database']
+    webserver = ansible_data['zabbix_web_http_server']
 
-    zabbix_web = host.package(f'zabbix-web-%s' % database)
+    zabbix_web = host.package(f'zabbix-%s-conf' % webserver)
     assert str(version) in zabbix_web.version
 
 

